lib/annotate/annotate_models.rb

```diff
@@ -908,6 +913,58 @@ def get_attributes(column, column_type, klass, options)
 
       attrs
     end
+
+    # Returns the regular expression for finding a column definition based on
+    # the format of the annotations being made
+    #
+    # == Returns:
+    # Regular expression
+    #
+    # @param [Symbol] symbol representation of the format being used for
+    #   annotations
+    def column_pattern_for(format_type)
+      case format_type
+      when :markdown then /^#\s+\*{2}`\w+`\*{2}(?:\s+\|\s`.*`)*$/
```

Comment:
I went for stricter matches. I'm not sure if it would make sense to be a bit looser and do something like:

```
/^#\s+\*{2}`\w+.*$/
```

that would catch 

```
# **column_name** ...
```

Currently, I'm checking for the fact there _may_ be additional columns in the markdown table

Reply:
After revisiting this PR, I'm not sure how I feel about this. I'm sure if we just grabbing all lines in the annotation section would end up being a problem. 

The previous implementation seemed to sort of be checking that the comment had an indication that it was the "column section" (e.g., `# col_name` or `# * col_name`).  This PR goes a bit further and does some stricter checking based on the format.

If we pulled in any non-empty comment line, it seems it _shouldn't_ cause a problem since we are essentially [checking for _any_ difference](https://github.com/ctran/annotate_models/pull/784/files#diff-2741ac3b166ed72003c60347ac3e3586R388). If we pulled in non-column comments, we'd get them in both `old_columns` and `new_columns`, so it shouldn't be flagged as a change. 

I _think_ the situation that we are avoiding by doing it this way is if there are non-annotation comments in the annotation block since it looks like the `header_pattern` grabs all groups comments after the `Table name:` comment.